ui/dashboards/src/components/AddPanel/AddPanel.tsx

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  MenuItem,
+  Stack,
+  Select,
+  TextField,
+  InputLabel,
+  FormControl,
+  Grid,
+  SelectChangeEvent,
+  Box,
+  Button,
+  Typography,
+} from '@mui/material';
+import { Drawer } from '@perses-dev/components';
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import { useLayouts, usePanels } from '../../context';
+
+interface AddPanelProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const AddPanel = ({ isOpen, onClose }: AddPanelProps) => {
+  const { layouts, addItemToLayout } = useLayouts();
+  const { addPanel } = usePanels();
+
+  const [group, setGroup] = useState(0);
+  const [panelName, setPanelName] = useState('');
+  const [panelDescription, setPanelDescription] = useState('');
+
+  const onSelectGroupChange = (e: SelectChangeEvent<number>) => {
+    setGroup(e.target.value as number);
+  };
+
+  const onPanelNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setPanelName(e.target.value);
+  };
+
+  const onPanelDescriptionChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setPanelDescription(e.target.value);
+  };
+
+  const onAddPanelClick = (e: FormEvent) => {
+    e.preventDefault();
+    addPanel(panelName, {
```

Comment:
Rather than using the raw panel name input, do you think it might be worthwhile to try and generate a "key" here that avoids using spaces and other special characters?

Reply:
i think in the future we also need to add validations, like check for duplicate panel name